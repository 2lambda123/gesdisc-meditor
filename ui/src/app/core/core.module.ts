import { NgModule } from '@angular/core';
import { CommonModule } from '@angular/common';
import { RouterModule } from '@angular/router';
import { FlexLayoutModule } from '@angular/flex-layout';
// CONTAINERS
import { MainComponent } from './containers/main';
import { NotFoundPageComponent } from './containers/not-found-page';
import { SplashPageContainer } from './containers/splash-page/splash-page.container';
// COMPONENTS
import { ContentTypeBoxComponent } from './components/content-type-button/content-type-button.component';
import { ContentTypeSplashComponent } from './components/splash-box/splash-box.component';
import { LayoutComponent } from './components/layout';
import { NavItemComponent } from './components/nav-item';
import { SidenavComponent } from './components/sidenav';
import { ToolbarComponent } from './components/toolbar';
import { MaterialModule } from '../material';

import { AngularFontAwesomeModule } from 'angular-font-awesome';

import { AuthModule } from '../auth/auth.module';

export const COMPONENTS = [
  MainComponent,
  NotFoundPageComponent,
  SplashPageContainer,
  ContentTypeBoxComponent,
  ContentTypeSplashComponent,
  LayoutComponent,
  NavItemComponent,
  SidenavComponent,
  ToolbarComponent,
];

@NgModule({
  imports: [CommonModule, RouterModule, MaterialModule, FlexLayoutModule, AngularFontAwesomeModule, AuthModule.forRoot()],
  declarations: COMPONENTS,
  exports: COMPONENTS
})

export class CoreModule {
  static forRoot() {
    return {
      ngModule: CoreModule
    };
  }
}
