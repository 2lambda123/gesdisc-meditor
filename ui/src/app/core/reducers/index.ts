import { createSelector, createFeatureSelector, ActionReducerMap } from '@ngrx/store';
import * as fromContentTypes from './content-types';

export interface ContentTypesState {
  contentTypes: fromContentTypes.State;
}

export interface State {
  contentTypes: ContentTypesState;
}

export const reducers = {
  contentTypes: fromContentTypes.reducer
};

export const selectContentTypesState = createFeatureSelector<ContentTypesState>('contentTypes');


export const getContentTypesEntitiesState = createSelector(
  selectContentTypesState,
  state => state.contentTypes
);

export const {
  selectIds: getContentTypeIds,
  selectEntities: getContentTypeEntities,
  selectAll: getAllContentTypes,
  selectTotal: getTotalContentTypes,
} = fromContentTypes.adapter.getSelectors(getContentTypesEntitiesState);
// export const selectContentTypesIds = createSelector(selectContentTypesState, fromContentTypes.selectContentTypesIds);
// export const selectContentTypesEntities = createSelector(selectContentTypesState, fromContentTypes.selectContentTypesEntities);
// export const selectAllContentTypes = createSelector(selectContentTypesState, fromContentTypes.selectAllContentTypes);
// export const selectContentTypesTotal = createSelector(selectContentTypesState, fromContentTypes.selectContentTypesTotal);
// export const selectCurrentUserId = createSelector(selectUserState, fromContentTypes.getSelectedUserId);

// export const selectCurrentUser = createSelector(
//   selectUserEntities,
//   selectCurrentUserId,
//   (userEntities, userId) => userEntities[userId]
// );
