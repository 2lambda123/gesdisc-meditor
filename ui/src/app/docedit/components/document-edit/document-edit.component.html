<json-schema-form *ngIf="schema && customWidgets"
	[schema] = "schema"
	[layout] = "layout"
	[data] = "data"
	[options]="jsonFormOptions"
	[framework]="selectedFramework"
	[loadExternalAssets]="true"
	[widgets]="customWidgets"
	(onChanges)="onChanges($event)"
	(onSubmit)="onSubmit($event)"
	(isValid)="isValid($event)"
	(validationErrors)="validationErrors($event)">
</json-schema-form>

