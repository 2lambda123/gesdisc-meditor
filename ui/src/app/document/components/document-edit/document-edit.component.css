.readonly-doc {
	pointer-events: none;
}