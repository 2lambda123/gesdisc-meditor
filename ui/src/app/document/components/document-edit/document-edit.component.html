<button mat-raised-button
 *ngIf="showExpandButton"
 (click)="toggleAllSections()" 
  style="margin-top: 1em;">{{ expandAll ? 'Expand all' : 'Collapse all' }}</button>

<div style="display:flex; margin-top: 30px;">
	<div #reactform style="flex: 1 0 auto;"></div>
</div>
