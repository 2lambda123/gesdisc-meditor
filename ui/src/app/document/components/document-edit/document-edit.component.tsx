import {
    Component,
    Input,
    Output,
    EventEmitter,
    ChangeDetectionStrategy,
    ViewChild,
    ElementRef,
    Renderer2,
} from '@angular/core'
import isEqual from 'lodash/isEqual' // lodash/isMatch does deep comparison, underscore/isMatch is shallow
import isEmpty from 'lodash/isEmpty'
import cloneDeep from 'lodash.clonedeep'
import * as _ from 'underscore'
// @ts-ignore
import React from 'react'
import ReactDOM from 'react-dom'
import Form from 'react-jsonschema-form'

const log = (type: any) => console.log.bind(console, type)

@Component({
    selector: 'med-document-edit',
    templateUrl: './document-edit.component.html',
    styleUrls: ['./document-edit.component.css'],
    changeDetection: ChangeDetectionStrategy.OnPush,
})
export class DocumentEditComponent {
    @ViewChild('reactform') reactformRef: ElementRef

    constructor(private renderer2: Renderer2) {}

    @Input()
    set document(document: any) {
        if (document.schema) {
            const schemaString = document.schema.replace("'", '')
            this.schema = JSON.parse(schemaString)
        }

        this.data = document.doc
        try {
            delete this.data.banTransitions
            delete this.data._id
        } catch (err) {}

        this.renderSchemaInForm()
    }

    @Input()
    set model(model: any) {
        this.uiSchema = model.uiSchema || '{}'
        this.renderSchemaInForm()
    }

    @Input() readonly: boolean

    @Output() liveData = new EventEmitter<object>()
    @Output() isValid = new EventEmitter<boolean>()
    @Output() isDirty = new EventEmitter<boolean>()

    schema = {}
    data: any
    uiSchema = '{}'

    formIsValid: boolean
    expandAll: boolean
    showExpandButton: boolean

    ngOnInit() {
        //remove react form Submit button

        let btns = this.reactformRef.nativeElement.querySelectorAll('.btn')
        let submit = btns[btns.length - 1]
        this.renderer2.removeChild(this.reactformRef.nativeElement, submit)
    }

    renderSchemaInForm() {
        if (!this.schema || !this.uiSchema) return

        console.log('render schema in form ')

        // make a copy of the form data, otherwise angular will trigger validation for every field
        // on every change
        let formData = cloneDeep(this.data)

        ReactDOM.render(
            React.createElement(
                Form,
                {
                    schema: this.schema,
                    formData,
                    uiSchema: JSON.parse(this.uiSchema),
                    liveValidate: true,
                    //widgets,
                    onChange: (e: any) => {
                        this.isFormValid(isEmpty(e.errors))
                        this.onChanges(e.formData)
                    },
                    onSubmit: (e: any) => {},
                    onError: log('errors'),
                },
                null
            ),
            this.reactformRef.nativeElement
        )
    }

    ngOnDestroy() {
        ReactDOM.unmountComponentAtNode(this.reactformRef.nativeElement)
    }

    onChanges(data: any) {
        Object.keys(data).forEach(key =>
            data[key] === undefined ? delete data[key] : ''
        )
        this.isDirty.emit(!isEqual(this.data, data))
        this.liveData.emit(data)
    }

    isFormValid(isvalid: boolean): void {
        this.isValid.emit(isvalid)
    }
}
