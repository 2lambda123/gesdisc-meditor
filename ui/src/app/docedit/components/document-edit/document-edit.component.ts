import { Component, OnInit, Input, Output, EventEmitter } from '@angular/core';
import { MaterialCkeditorComponent } from '../../widgets/material-ckeditor.component';

@Component({
  selector: 'med-document-edit',
  templateUrl: './document-edit.component.html',
  styleUrls: ['./document-edit.component.css']
})

export class DocumentEditComponent implements OnInit {

	@Input()
  set document(document: any) {
    if(document.document.schema) {
    	let schemaString = document.document.schema.replace('\'', '');
	    this.schema = JSON.parse(schemaString).properties;
	    this.data = document.document.doc;
    }
  }

  @Output() submitDocument = new EventEmitter<object>();

	selectedFramework = "material-design";
	jsonFormOptions = {};
	schema =  {};
	data = {};
	layout = [
		"abstract",
		"expiration",
		"start",
		"severity",
		// "body"
		{ "key": "body", "widget" : "ckeditor" }
	];

	customWidgets = {
	  ckeditor: MaterialCkeditorComponent,
	}


	submittedFormData = {};
	liveFormData = {};
	formValidationErrors = {};
	formIsValid: boolean;


  ngOnInit() {

  }

  onSubmit() {
    this.submitDocument.emit(this.data);
  }

  onChanges(data: any) {
    this.liveFormData = data;
  }

  isValid(isValid: boolean): void {
    this.formIsValid = isValid;
  }

  validationErrors(data: any): void {
    this.formValidationErrors = data;
  }

}
