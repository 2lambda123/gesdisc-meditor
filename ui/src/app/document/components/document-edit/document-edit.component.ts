import { Component, OnInit, Input, Output, EventEmitter } from '@angular/core';

@Component({
	selector: 'med-document-edit',
	templateUrl: './document-edit.component.html',
	styleUrls: ['./document-edit.component.css']
})

export class DocumentEditComponent implements OnInit {

	@Input()
	set document(document: any) {
		if (document.schema) {
			const schemaString = document.schema.replace('\'', '');
			this.schema = JSON.parse(schemaString);
		}

		if (document.layout) {
			const layoutString = document.layout.replace('\'', '');
			this.layout = JSON.parse(layoutString);
		}

		this.data = document.doc;
	}

	@Output() liveData = new EventEmitter<object>();
	@Output() isValid = new EventEmitter<boolean>();

	selectedFramework = "material-design";
	jsonFormOptions = {
		addSubmit: false
	};
	schema =  {};
	data = {};
	layout = undefined;
	defaultLayoutOptions = {
		fxLayoutGap: '50px',
	};

	submittedFormData = {};
	liveFormData = {};
	formValidationErrors = {};
	formIsValid: boolean;


	ngOnInit() {

	}

	// onSubmit() {
	// 	this.submitDocument.emit(this.liveFormData);
	// }

	onChanges(data: any) {
		// this.liveFormData = data;
		this.liveData.emit(data);
	}

	isFormValid(isvalid: boolean): void {
		// this.formIsValid = isvalid;
		this.isValid.emit(isvalid);
	}

	validationErrors(data: any): void {
		this.formValidationErrors = data;
	}

}
