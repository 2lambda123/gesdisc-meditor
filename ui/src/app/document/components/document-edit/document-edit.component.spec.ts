import { async, ComponentFixture, TestBed } from '@angular/core/testing';
import { NO_ERRORS_SCHEMA } from '@angular/core';

import { DocumentEditComponent } from './document-edit.component';

describe('DocumentEditComponent', () => {
  let component: DocumentEditComponent;
  let fixture: ComponentFixture<DocumentEditComponent>;

  beforeEach(async(() => {
    TestBed.configureTestingModule({
      declarations: [ DocumentEditComponent ],
      schemas: [ NO_ERRORS_SCHEMA ]
    })
    .compileComponents();
  }));

  beforeEach(() => {
    fixture = TestBed.createComponent(DocumentEditComponent);
    component = fixture.componentInstance;
  });

  it('should create', () => {
    fixture.detectChanges();
		expect(fixture).toMatchSnapshot();
  });
});