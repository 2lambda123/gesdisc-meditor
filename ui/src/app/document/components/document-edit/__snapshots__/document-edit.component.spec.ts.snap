// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DocumentEditComponent should create 1`] = `
<med-document-edit
  data={[Function Object]}
  defaultLayoutOptions={[Function Object]}
  formValidationErrors={[Function Object]}
  isDirty={[Function EventEmitter]}
  isValid={[Function EventEmitter]}
  jsonFormOptions={[Function Object]}
  layout="undefined"
  liveData={[Function EventEmitter]}
  schema={[Function Object]}
  selectedFramework={[Function String]}
  submittedFormData={[Function Object]}
>
  <json-schema-form />
</med-document-edit>
`;
